storage_service/locations/api/beta/remple/openapi.py

```diff
@@ -0,0 +1,1573 @@
+                ('paginator', {'$ref': self._get_paginator_schema_path()}),
+            ])),
+            ('required', ['query']),
+        ])
+        return search_schema_name, search_schema
+
+    def _get_new_search_schema(self, resource_name, resource_cfg, read_schema):
+        new_search_schema_name = self._get_new_search_schema_name(resource_name)
+        new_search_schema = OrderedDict([
+            ('type', 'object'),
+            ('properties', OrderedDict([
+                ('search_parameters', OrderedDict([
+                    # TODO: this is not a string, but an object with
+                    # "attributes" and "relations".
+                    ('type', 'string'),
+                ])),
+            ])),
+            ('required', ['search_parameters']),
+        ])
+        return new_search_schema_name, new_search_schema
+
+    def _get_new_schema(self, resource_name, resource_cfg, read_schema):
+        new_schema_name = self._get_new_schema_name(resource_name)
+        new_schema = OrderedDict([('type', 'object')])
+        properties = OrderedDict()
+        resource_cls = resource_cfg['resource_cls']
+        resource_inst = resource_cls(
+            None, server_path=self.get_dflt_server_path(),
+            other_resources=self.resources)
+        required_fields = []
+        for field_name in resource_inst._get_new_edit_collections():
+            properties[field_name] = OrderedDict([
+                ('type', 'array'),
+                ('items', OrderedDict([
+                    ('type', 'string'),
+                    ('format', 'URI of a(n) {} resource'.format(field_name)),
+                ])),
+            ])
+            required_fields.append(field_name)
+        if properties:
+            new_schema['properties'] = properties
+            new_schema['required'] = required_fields
+        return new_schema_name, new_schema
+
+    def _get_create_schema(self, resource_name, resource_cfg, read_schema):
+        """Return a create schema for the resource named ``resource_name``.
+
+        The create schema describes what is needed to create an instance of the
+        input-named resource.
+        """
+        create_schema_name = self._get_create_schema_name(resource_name)
+        resource_cls = resource_cfg['resource_cls']
+        schema_cls = resource_cls.get_create_schema_cls()
+        create_schema = self._get_create_update_schema(
+            resource_name, resource_cfg, read_schema, schema_cls)
+        return create_schema_name, create_schema
+
+    def _get_update_schema(self, resource_name, resource_cfg, read_schema):
+        """Return an update schema for the resource named ``resource_name``.
+
+        The update schema describes what is needed to update an instance of the
+        input-named resource.
+
+        TODO: should every parameter in an update request be optional, given
+        that the resource being updated is presumably valid?
```

Comment:
No. This comment is misleading. Update via Remple has the semantics of `PUT` not `PATCH`. A full representation of the post-update state must be sent in the update request.